Pcap++/src/DpdkDeviceList.cpp

```diff
@@ -400,6 +400,54 @@ bool DpdkDeviceList::startDpdkWorkerThreads(CoreMask coreMask, std::vector<DpdkW
 	return true;
 }
 
+bool DpdkDeviceList::startDpdkWorkerThread(CoreMask coreMask, DpdkWorkerThread* worker)
+{
+	if (!isInitialized())
+	{
+		PCPP_LOG_ERROR("DpdkDeviceList not initialized");
+		return false;
+	}
+
+	CoreMask tempCoreMask = coreMask;
+	size_t numOfCoresInMask = 0;
+	int coreNum = 0;
+	while (tempCoreMask > 0)
+	{
+		if (tempCoreMask & 1)
+		{
+			if (!rte_lcore_is_enabled(coreNum))
+			{
+				PCPP_LOG_ERROR("Trying to use core #" << coreNum << " which isn't initialized by DPDK");
+				return false;
+			}
+
+			numOfCoresInMask++;
+		}
+		tempCoreMask = tempCoreMask >> 1;
+		coreNum++;
+	}
+
+	if (numOfCoresInMask == 0)
+	{
+		PCPP_LOG_ERROR("Number of cores in mask is 0");
+		return false;
+	}
+	m_WorkerThreads.clear();
+        int lcore_id,ret =0;
+        rte_eal_mp_remote_launch(dpdkWorkerThreadStart,worker, CALL_MASTER);
```

Comment:
In https://github.com/seladb/PcapPlusPlus/issues/1099#issuecomment-1475473452 you said that the master core blocks the main thread, is it addressed here somehow? 🤔

Reply:
I started a new thread inside the main thread callback function